src/internet_identity/src/assets.rs

```diff
@@ -6,9 +6,33 @@ use crate::{http, state};
 use base64::engine::general_purpose::STANDARD as BASE64;
 use base64::Engine;
 use ic_cdk::api;
+use ic_certified_map::{AsHashTree, Hash, HashTree, RbTree};
 use include_dir::{include_dir, Dir, File};
+use internet_identity_interface::http_gateway::HeaderField;
 use lazy_static::lazy_static;
 use sha2::Digest;
+use std::collections::HashMap;
+
+const LABEL_ASSETS: &[u8] = b"http_assets";
+
+#[derive(Debug, Default, Clone)]
+pub struct CertifiedAssets {
+    pub assets: HashMap<String, (Vec<HeaderField>, Vec<u8>)>,
+    pub certification_v1: RbTree<String, Hash>,
+}
+
+impl CertifiedAssets {
+    /// Returns the root_hash of the asset certification tree.
+    /// In the future, this will include both certification v1 and v2.
```

Comment:
nit: not sure we need this remark about the future -- it doesn't contribute to explaining the current state of the code.  If you prefer to keep it, maybe turn into a TODO?